src/Symfony/Component/Semaphore/Semaphore.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Semaphore;
+
+use Psr\Log\LoggerAwareInterface;
+use Psr\Log\LoggerAwareTrait;
+use Psr\Log\NullLogger;
+use Symfony\Component\Semaphore\Exception\InvalidArgumentException;
+use Symfony\Component\Semaphore\Exception\RuntimeException;
+use Symfony\Component\Semaphore\Exception\SemaphoreAcquiringException;
+use Symfony\Component\Semaphore\Exception\SemaphoreExpiredException;
+use Symfony\Component\Semaphore\Exception\SemaphoreReleasingException;
+
+/**
+ * Semaphore is the default implementation of the SemaphoreInterface.
+ *
+ * @experimental in 5.2
+ *
+ * @author Grégoire Pineau <lyrixx@lyrixx.info>
+ * @author Jérémy Derussé <jeremy@derusse.com>
+ */
+final class Semaphore implements SemaphoreInterface, LoggerAwareInterface
+{
+    use LoggerAwareTrait;
+
+    private $store;
+    private $key;
+    private $ttlInSecond;
+    private $autoRelease;
+    private $dirty = false;
+
+    public function __construct(Key $key, PersistingStoreInterface $store, float $ttlInSecond = 300.0, bool $autoRelease = true)
+    {
+        $this->store = $store;
+        $this->key = $key;
+        $this->ttlInSecond = $ttlInSecond;
+        $this->autoRelease = $autoRelease;
+
+        $this->logger = new NullLogger();
+    }
+
+    /**
+     * Automatically releases the underlying semaphore when the object is destructed.
+     */
+    public function __destruct()
+    {
+        if (!$this->autoRelease || !$this->dirty || !$this->isAcquired()) {
+            return;
+        }
+
+        $this->release();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function acquire(): bool
+    {
+        try {
+            $this->store->save($this->key, $this->ttlInSecond);
+            $this->key->reduceLifetime($this->ttlInSecond);
+            $this->dirty = true;
+
+            $this->logger->info('Successfully acquired the "{resource}" semaphore.', ['resource' => $this->key]);
```

Comment:
reduce verbosity (same for other logs. see recent version of lock for reference)